File: src/main/scala/org/anz/codechallenge/validators/FileNameValidator.scala
package org.anz.codechallenge.validators

import com.google.gson.Gson
import org.anz.codechallenge.schema.{Metadata, Schema, Tag}
import org.apache.spark.sql.DataFrame
import org.apache.spark.sql.functions.col

import scala.io.Source

/**
  * File name validator
  * @param inputMetadata - Input metadata representing metadata info of input file
  * @param file_schema - File schema representing schema of the input file
  * @param tagFile - Tagfile representing he file name and record count
  * @param dataFrame - Input file read as dataframe which needs validation
  */
class FileNameValidator (
                          inputMetadata: Metadata,
                          file_schema: Schema,
                          tagFile: Tag,
                          dataFrame: DataFrame
                        ) extends Validator {

  /**
    * Perform file name validation
    * @return - status of validation
    */
  override def validate(): String = {
    println("Performing file name validation")

    val expectedFilename = tagFile.file_name

    println("expected file name is "+expectedFilename)

    val status = if(inputMetadata.data.endsWith(expectedFilename))  "0" else "2"

    println("File name validation status is "+status)
    status
  }
}

File: src/main/java/org/anz/codechallenge/validators/FileNameValidator.java
package org.anz.codechallenge.validators;

import org.anz.codechallenge.filedetails.ContentParams;
import org.anz.codechallenge.filedetails.FileContent;
import org.anz.codechallenge.schema.JSONSchema;
import org.anz.codechallenge.tags.DelimitedTag;
import org.anz.codechallenge.tags.Tag;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;

public class FileNameValidator implements Validator {

    private FileContent fileContent;

    public FileNameValidator(FileContent fileContent) {
        this.fileContent = fileContent;
    }

    @Override
    public String validate() {
        System.out.println("Performing file name validation");

        Tag tagFile = fileContent.getFileMetadata().getTagFile();
        String expectedFilename = "";
        if(tagFile instanceof DelimitedTag) {
            DelimitedTag delimitedTag = (DelimitedTag) tagFile;
            expectedFilename = delimitedTag.getFile_name();

            System.out.println("expected file name is " + expectedFilename);
        }
        String status = (fileContent.getFileMetadata().getFilePath().endsWith(expectedFilename)) ? "0" : "2";

        System.out.println("File name validation status is "+status);
        return status;
    }
}
